staging/src/k8s.io/externaljwt/pkg/plugin/plugin.go

```diff
@@ -0,0 +1,268 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package plugin
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"gopkg.in/square/go-jose.v2/jwt"
+	ejspb "k8s.io/externaljwt/apis/v1alpha1"
+)
+
+type PublicKey struct {
+	KeyID                    string
+	PublicKey                interface{}
+	ExcludeFromOIDCDiscovery bool
+}
+
+type VerificationKeys struct {
+	Keys          []PublicKey
+	DataTimestamp time.Time
+}
+
+type Metadata struct {
+	MaxTokenExpirationSeconds int
+}
+
+// New sets up all dependencies for the plugin and cache, in
```

Comment:
This sounds like a comment from an internal fork, please fix.

Reply:
Done